zpeclient.go

```diff
@@ -0,0 +1,227 @@
+package webhook
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	v1 "github.com/yahoo/k8s-athenz-istio-auth/pkg/apis/athenz/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	replacer     = strings.NewReplacer(".*", ".*", "*", ".*")
+	privateCache *Cache
+)
+
+// roleMappings - cr cache
+type roleMappings struct {
+	roleToPrincipals map[string][]*simplePrincipal
+	roleToAssertion  map[string][]*simpleAssertion
+}
+
+// simplePrincipal - principal data
+type simplePrincipal struct {
+	memberRegex *regexp.Regexp
+	expiration  time.Time
+}
+
+// simpleAssertion - processed policy
+type simpleAssertion struct {
+	resource *regexp.Regexp
+	action   *regexp.Regexp
+	effect   string
+}
+
+// Cache - cache for athenzdomains CR
+type Cache struct {
+	CrIndexInformer cache.SharedIndexInformer
+	DomainMap       map[string]roleMappings
+	lock            sync.RWMutex
+}
+
+// BuildCache - generate new athenzdomains cr cache
+func BuildCache(crIndexInformer cache.SharedIndexInformer) {
```

Comment:
Can we rename to function to NewZpeClient which returns a cache object which has an authorize function exposed?